File: DFS.cpp
#include<bits/stdc++.h>
using namespace std;
vector<vector<int> > adj;
bool vis[100005];
void dfs(int x){
    vis[x]= true;
    for(int i=0;i<adj[x].size();i++){
        int u = adj[x][i];
        if(!vis[u])
           dfs(u);
    }
}
int main(){
    int N,M;
    cin>>N>>M;
    adj.resize(N+1);
    memset(vis,false, N+1);
    while(M--){
        int u,v;
        cin>>u>>v;
        adj[u].push_back(v);
        adj[v].push_back(u);
    }
    int x;
    cin>>x;
    dfs(x);
    int cnt =0;
    for(int i=1;i<=N;i++){
        if(!vis[i])
        cnt++;
    }
    cout<<cnt<<endl;
}

File: Dfs.java

import java.io.BufferedReader;
import java.io.InputStreamReader;

import java.util.*;

class Graph {
    private int v;
    public List<Integer> adj[];
    public Graph(int n){
        v= n;
      adj = new ArrayList[n];
      for(int i=0;i<adj.length;i++){
          adj[i] = new ArrayList<Integer>();
      }
    }
    public void addEdge(int u,int v){
        adj[u].add(v);
    }
    public void dfs(int u,boolean vis[]){
        vis[u] = true;
        for(int v : adj[u]){
          if(!vis[v]){
              dfs(v,vis);
          }
        }
    }
}

public class TestClass {
    
    public static void main(String args[] ) throws Exception {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        int m = sc.nextInt();
        Graph g = new Graph(n+1);
        for(int i=0;i<m;i++){
            int u = sc.nextInt();
            int v = sc.nextInt();
            g.addEdge(u,v);
            g.addEdge(v,u);
        }
        int origin = sc.nextInt();
        boolean[] vis = new boolean[n+1];
        Arrays.fill(vis,false);
        g.dfs(origin,vis);
        int cnt = 0;
        for(int i=1;i<=n;i++){
            if(vis[i]==false)
            cnt++;
        }
        System.out.println(cnt);;
    }

}
